### DISCUSSION TITLE
What are the differences between Termux enviroment and Android ?

### DISCUSSION THREAD

**Discussion by gorgeously-app** [termux/termux-app#2459] [Q&A] [answered]
I'm trying to run some executable using Runtime,exec directly vs using Termux. Some only work with Termux although the exact same build (built from source for android).
Can someone explain the differences ?
Thanks!

**Comment by agnostic-apollo** [comment, marked as answer]
> some executable

Which executables?

Termux binaries are specifically compiled for termux app with the package/`$PREFIX` `com.termux`. They cannot be used in other apps. They need to be compiled for each app.

https://github.com/termux/termux-app#Forking

**Reply by gorgeously-app** [reply]
Thanks for your answer ! I'm just testing things out to learn more about Termux. They are just some random github sources I first built with termux using cmake and copy to use in other apps or adb shell to see if they can work. All of them have this error: 
"syntax error: unexpected '('"

Are there any quick fix to build from source with cmake or else and make these binaries usable for other android apps ? Or forking is the only solution ?

Sorry if I misused the terms.

**Reply by agnostic-apollo** [reply]
>They are just some random github sources I first built ... All of them have this error: "syntax error: unexpected '('"

Without more info on what you are compiling and whether such stuff is termux and/or architecture independent and the steps you are performing, we cannot help you. Moreover, we only provide support for compilation issues for packages officially supplied by termux, like via `termux-packages` repo and **do not** provide support for compilation issues of "random github sources", you should create issues in their repos or ask for help in unofficial termux forums.

>Are there any quick fix to build from source with cmake or else and make these binaries usable for other android apps ? Or forking is the only solution ?

You don't need to fork `termux-app` but you do need to fork `termux-packages` repo and modify [`properties.sh`](https://github.com/termux/termux-packages/blob/master/scripts/properties.sh) with respective app's package name and then [build all packages](https://github.com/termux/termux-packages/wiki/Building-packages), optionally static builds. Then extract the executables in app's private app data directory. There will be execution restrictions as well if that app uses `targetSdkVersion` `> 28` and you will need something like `updated-flavors` branch mechanism to run exectuables from app's APK native lib directory. There is no other way, depending on package requirements/patches.

https://github.com/termux/termux-packages/wiki/Termux-and-Android-10
